sklearn/kernel_ridge/tests/test_krc.py

```diff
@@ -0,0 +1,203 @@
+"""
+Testing for Kernel Ridge Classification
+(sklearn.kernel_ridge)
+
+Authors: Carlos Perales <sir.perales@gmail.com>
+License: BSD 3 clause
+
+"""
+import numpy as np
+from numpy.testing import assert_array_equal, assert_array_almost_equal
+from numpy.testing import assert_almost_equal
+from sklearn import datasets, base
+from sklearn.utils import check_random_state
+from sklearn.utils.testing import assert_equal, assert_true
+from sklearn.utils.testing import assert_greater
+from sklearn.utils.testing import assert_raises
+from sklearn.externals import six
+from sklearn.utils.estimator_checks import check_estimator
+from sklearn.kernel_ridge import KRC
+
+# toy sample
+X = [[-2, -1], [-1, -1], [-1, -2], [1, 1], [1, 2], [2, 1]]
+Y = [1, 1, 1, 2, 2, 2]
+T = [[-1, -1], [2, 2], [3, 2]]
+true_result = [1, 2, 2]
+
+# also load the iris dataset
+iris = datasets.load_iris()
+rng = check_random_state(42)
+perm = rng.permutation(iris.target.size)
+iris.data = iris.data[perm]
+iris.target = iris.target[perm]
+
+
+def test_check_estimator():
+    check_estimator(KRC)
+
+
+def test_krc_parameters():
+    # Test parameters.
+    clf = KRC(kernel='rbf').fit(X, Y)
+    assert_array_equal(clf.predict(X), Y)
+
+
+def test_precomputed():
```

Comment:
Surely we can share a test like this between KRR and KRC, without duplicating the code. this is also not very easy to read as a test -- I'm sure it could be simpler, but don't see the point in detailed comments until it is refactored with respect to KKR tests.

Reply:
Well, as far as I have learned from sklearn tests, a lot of tests are focus to the estimator you are testing, instead of the object itself.

So, Kernel Ridge Regressor and Kernel Ridge Classifier are tested in different ways, because, although they share same mathematical concepts and formulations, first one is used for predicting continue targets and second one for a discrete target, a number of classes.

Reply:
Generic estimator API is tested in common tests.

Tests are often written like some kind of proof by induction. You show a basic feature - like the ability to classify - works, then you show that modifying some parameter works too. Having shown that you can predict from `dual_coef_`, showing that `dual_coef_` is calculated correctly as a function of `kernel` is  sufficient to suggest that all will work together.

Reply:
Hmm, I see.. So what you are saying is that some tests are superfluous, because same property is tested twice, one as a regressor and one as a classifier, right?

I'll take a look today and I'll try to reduce the number of tests then.